### PR TITLE
Return an error if the signed tree root is not, in fact, signed

### PR DESCRIPTION
**Pull request by RJPercival**
This could happen if there is a problem generating a signature server-side. The server really shouldn't be sending a STR without a signature, but it currently does if a signing error occurs.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -120,6 +120,10 @@ func (c *LogClient) UpdateSTR(ctx context.Context) error {
 	}
 	str := resp.SignedLogRoot
 
+	if str.Signature == nil {
```
Comment (gdbelvin) [inline client/client.go]: Please move this check to be inside `crypto.Verify`

Comment (RJPercival) [inline client/client.go]: Done.

**Comment by gdbelvin** [comment]
Consider adding a nil test to the signature tests as well please.

**Comment by RJPercival** [comment]
Done.

**Pull request merged**